cmake/modules/DarwinSDKs.cmake

```diff
@@ -4,17 +4,34 @@ option(SWIFT_ENABLE_IOS32
 
 if(SWIFT_ENABLE_IOS32)
 set(SUPPORTED_IOS_ARCHS "armv7;armv7s;arm64;arm64e")
-set(SUPPORTED_IOS_SIMULATOR_ARCHS "i386;x86_64")
+set(SUPPORTED_IOS_SIMULATOR_ARCHS "i386;x86_64;arm64")
 else()
 set(SUPPORTED_IOS_ARCHS "arm64;arm64e")
-set(SUPPORTED_IOS_SIMULATOR_ARCHS "x86_64")
+set(SUPPORTED_IOS_SIMULATOR_ARCHS "x86_64;arm64")
 endif()
 
 set(SUPPORTED_TVOS_ARCHS "arm64")
-set(SUPPORTED_TVOS_SIMULATOR_ARCHS "x86_64")
+set(SUPPORTED_TVOS_SIMULATOR_ARCHS "x86_64;arm64")
 set(SUPPORTED_WATCHOS_ARCHS "armv7k")
-set(SUPPORTED_WATCHOS_SIMULATOR_ARCHS "i386")
-set(SUPPORTED_OSX_ARCHS "x86_64")
+set(SUPPORTED_WATCHOS_SIMULATOR_ARCHS "i386;arm64")
+set(SUPPORTED_OSX_ARCHS "x86_64;arm64;arm64e")
+
+# Get the SDK version from SDKSettings.
+execute_process(
+  COMMAND "defaults" "read" "${CMAKE_OSX_SYSROOT}/SDKSettings.plist" "Version"
+    OUTPUT_VARIABLE SWIFT_OSX_SDK_VERSION
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+# Remove the last component, if any. e.g. 10.15.26 -> 10.15
+string(REGEX REPLACE "\([0-9]*[.][0-9]*\)[.][0-9]*" "\\1"
+       SWIFT_OSX_SDK_VERSION "${SWIFT_OSX_SDK_VERSION}")
+
+if (${SWIFT_OSX_SDK_VERSION} STREQUAL "10.14" OR
+    ${SWIFT_OSX_SDK_VERSION} STREQUAL "10.15")
+  set(SUPPORTED_OSX_ARCHS "x86_64")
+else()
+  set(SUPPORTED_OSX_ARCHS "x86_64;arm64e")
```

Comment:
Should this include arm64 like line 17?

Reply:
Going to file a bugs.swift.org and follow up in the future PR.

Reply:
This bit of code should be dead now